### PR TITLE
Giving tuesday web

### PR DESCRIPTION
**Pull request by timosville**
Card: https://trello.com/c/D0rSDWrm/1271-giving-tuesday-web

image

image

### CODE REVIEW

File: src/assets/js/promo.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+dayjs.extend(isSameOrBefore);
+dayjs.extend(isSameOrAfter);
+dayjs.extend(customParseFormat);
+
+export default () => {
+  return {
+    startDate: '',
+    endDate: '',
+    now: dayjs(),
+    isPromoActive: false,
```
Comment (JannieT) [inline src/assets/js/promo.js:15]: this name is a bit deceptive. How about `hasDeclinedPromo: false`

File: src/assets/js/promo.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    async init(startDate, endDate) {
```
Comment (JannieT) [inline src/assets/js/promo.js:17]: not sure why this and `setPromoDates` is async.

**Review by JannieT — APPROVED** [review]
VERY well crafted solution! Two small suggestions for improvement.

File: src/assets/js/promo.js (hunk 2 above)
Comment (timosville) [reply src/assets/js/promo.js:17]: This stops it from firing twice on page load.

File: src/assets/js/promo.js (hunk 1 above)
Comment (timosville) [reply src/assets/js/promo.js:15]: Good catch. I will make the changes.

Comment (timosville) [reply src/assets/js/promo.js:15]: Renamed to `hasSeenPromo`

**Pull request merged**